packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -122,11 +123,26 @@ export class HostedPluginProcess implements ServerPluginRunner {
         const hostedPluginManager = rpc.getProxy(MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT);
         hostedPluginManager.$stopPlugin('').then(() => {
             emitter.dispose();
-            cp.kill();
+            this.killProcessTree(cp);
         });
 
     }
 
+    private killProcessTree(parentProcess: cp.ChildProcess): void {
+        parentProcess.disconnect();
+        const parentPid = parentProcess.pid;
+        if (isWindows) {
+            cp.spawn('taskkill', ['/F', '/T', '/PID', parentPid.toString()]);
```

Comment:
I had a very similar issue in one of my downstream projects. A few remarks on the PR:
 - Instead of doing `taskkill` on Windows and `kill` on *nix, just use `tree-kill`:
```ts
require('tree-kill')(yourPID);
```
 - You have to get the process tree on Windows as well. Not just in the `else` branch.

Reply:
interesting that VS Code uses `taskkill` for it: https://github.com/microsoft/vscode/blob/a848f18df4efa9eb947bd01cbebcede418e66148/src/vs/workbench/contrib/debug/node/debugAdapter.ts#L269-L275

Reply:
ah, i see tree-kill does it https://github.com/pkrumins/node-tree-kill/blob/master/index.js#L20

Reply:
Thanks for your recommendation, I will try to use this new npm package.

Reply:
@tom-shan, please ping if you are done, I'll try to find time for the verification. Thanks!